### ISSUE TITLE
Automatize the metadata parsing

### ISSUE THREAD

**Issue by urinieto** [urinieto/msaf-data#5]
We should have a global file containing all the metadata in order to avoid having to manually add it after parsing the svl files. Thus, the parser could add it automatically, instead of destroying it (like in PR #2).

**Comment by urinieto** [comment]
Ok, the [parser](https://github.com/urinieto/msaf-data/blob/master/SPAM/scripts/parse_annotations.py) now includes the metadata information, extracted from this new [tsv file](https://github.com/urinieto/msaf-data/blob/master/SPAM/metadata.tsv). However, after I used @bmcfee 's [notebook](https://github.com/urinieto/msaf-data/blob/master/SPAM/scripts/SPAM%20annotation%20correction.ipynb) to correct the annotations, I get a series of errors for some files (eg, the famous SALAMI_114) and it doesn't really include any multisegment annotations for the ones that fail.

@bmcfee , did you do anything else to the JAMS files after using the [annotations parser](https://github.com/urinieto/msaf-data/blob/master/SPAM/scripts/parse_annotations.py) and before using your notebook to obtain the references you included in #2?

**Comment by bmcfee** [comment]
> I get a series of errors for some files (eg, the famous SALAMI_114) and it doesn't really include any multisegment annotations for the ones that fail.

That's correct.  The notebook does not produce multi-segment annotations if the upper and lower annotations cannot be resolved.

> did you do anything else

nope!

**Comment by urinieto** [comment]
So, how come the files you uploaded and fail contain multi_segment annotations (e.g. SALAMI_114 from your last [commit](https://github.com/urinieto/msaf-data/blob/a80d746e66b37b65cb2f5fc8e196ff80fd511e4f/SPAM/references/SALAMI_114.jams)) and mine don't (e.g., the same from my last [commit](https://github.com/urinieto/msaf-data/blob/master/SPAM/references/SALAMI_114.jams))?

**Comment by bmcfee** [comment]
O_o really?  do you have the most recent version of the notebook?

**Comment by urinieto** [comment]
Yeah, I just tried it again to confirm with [this version](https://gist.github.com/bmcfee/ec40f7ae72238bc1c95bc2ba95a9f785) and I get the same results. Not sure what's going on... can you try it yourself? This is what I did from the `SPAM/scripts` folder:

```
python parse_annotations.py ../original_references out_dir ../metadata.tsv
```

And then run the notebook on the `out_dir` folder.

**Comment by bmcfee** [comment]
> So, how come the files you uploaded and fail contain multi_segment annotations

Derp, sorry, I was unclear before.  Failure is determined on an annotator-basis, not track basis.  I still want the multi-level annotations for the annotators that can be resolved together, even for 114.  There should be four multi-level annotations in 114, five in all other tracks.

**Comment by urinieto** [comment]
Yeah, I get that. But the notebook doesn't do that, right? How did you
compute that in your last commit? I'm trying to automatize the whole thing

**Comment by bmcfee** [comment]
> Yeah, I get that. But the notebook doesn't do that, right? How did you
> compute that in your last commit?

Doesn't it?  My last commit was done just as you describe: run the parser script and then the notebook.

> I'm trying to automatize the whole thing

I think the best way to do this is to restructure the parser so that it collects all the annotations for a track first and then writes out the results.  Doing it piecemeal and operating in place makes it really difficult to keep synchronized.

**Comment by bmcfee** [comment]
... if you're already in the process of automating this step, how about taking a crack at the salami annotations as well?

I tried running the notebook over them, and they're a total mess. Maybe this is due to parse errors from the upstream annotations?

**Comment by urinieto** [comment]
> ... if you're already in the process of automating this step, how about taking a crack at the salami annotations as well?

I haven't started yet, since I still don't know how you actually computed the multi_segment. Can you try it again with the latest version of the parser + notebook and see if we get the same results? Once I get the same results you got, I will redo the parser such that it collects all the annotations for a track first.

> I tried running the notebook over them, and they're a total mess. Maybe this is due to parse errors from the upstream annotations?

Ugh... no idea, haven't tried the SALAMI ones
